lightningd/plugin.c

```diff
@@ -314,7 +363,7 @@ static struct io_plan *plugin_read_json(struct io_conn *conn UNUSED,
 	/* Read and process all messages from the connection */
 	do {
 		success = plugin_read_json_one(plugin);
-	} while (success);
+	} while (success && ! plugin->stop);
 
 	if (plugin->stop)
 		return io_close(plugin->stdout_conn);
```

Comment:
Good (unrelated) catch; but just check plugin->stop inside the loop, rather than doing so twice?